sdk/core/ts-http-runtime/src/policies/auth/apiKeyAuthenticationPolicy.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import type { ApiKeyCredential } from "../../auth/credentials.js";
+import type { AuthScheme } from "../../auth/schemes.js";
+import type { PipelineRequest, PipelineResponse, SendRequest } from "../../interfaces.js";
+import type { PipelinePolicy } from "../../pipeline.js";
+import { ensureSecureConnection } from "./checkInsecureConnection.js";
+
+/**
+ * Name of the API Key Authentication Policy
+ */
+export const apiKeyAuthenticationPolicyName = "apiKeyAuthenticationPolicy";
+
+/**
+ * Options for configuring the API key authentication policy
+ */
+export interface ApiKeyAuthenticationPolicyOptions {
+  /**
+   * The credential used to authenticate requests
+   */
+  credential: ApiKeyCredential;
+  /**
+   * Optional authentication schemes to use. If not provided, schemes from the request will be used.
```

Comment:
this makes it sound like authSchemes in the policy options has precedence, but looking below we default to using the request authSchemes and then fallback to the policy options

Reply:
Updated the comment!